azurerm/resource_arm_virtual_network_gateway_connection.go

```diff
@@ -294,14 +420,27 @@ func getArmVirtualNetworkGatewayConnectionProperties(d *schema.ResourceData) (*n
 		}
 	}
 
+	if v, ok := d.GetOk("enable_bgp"); ok {
```

Comment:
if this is a bool, it'll always have a value (since the default is `false`) - as such we may as well just assign this above?

Reply:
The idea was to just not set the field in the API struct, if it is not defined in the Terraform schema instance. Is this overly safe?

Reply:
It's more this won't work as it reads unfortunately due to the way the defaults work in HCL (there's changes coming in this area in the future); given the field in the schema is a Boolean, there'll always be a default value of `false`, so the `ok` in this if statement will always be true.

We _could_ instead update this to use `GetOkExists` which returns the raw value instead of the defaulted value if nothings specified (e.g. an empty string in this case, iirc) - however I don't think there's any harm setting a default to `false` tbh - what do you think?